tests/testutils/goroutineassistant.go

```diff
@@ -18,23 +18,25 @@ import (
 	"fmt"
 	"sync"
 	"testing"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/steveeJ/gexpect"
 )
 
 type GoroutineAssistant struct {
-	s  chan string
+	s  chan error
 	wg sync.WaitGroup
 	t  *testing.T
 }
 
 func NewGoroutineAssistant(t *testing.T) *GoroutineAssistant {
 	return &GoroutineAssistant{
-		s: make(chan string),
+		s: make(chan error),
 		t: t,
 	}
 }
 
 func (a *GoroutineAssistant) Fatalf(s string, args ...interface{}) {
-	a.s <- fmt.Sprintf(s, args...)
+	a.s <- fmt.Errorf(s, args...)
```

Comment:
Sending to this channel is usually the last action goroutine does explicitly (that is - not counting the deferred actions).

I wanted to have a buffered channel, just to let the goroutine to die quickly, without waiting for the receiving side of channel, so deferred actions can be executed immediately.